runtime/include/wasm_stack.h

```diff
@@ -21,48 +28,93 @@ struct wasm_stack {
  * @returns 0 on success, -1 on error
  */
 static inline int
-wasm_stack_allocate(struct wasm_stack *stack, size_t capacity)
+wasm_stack_init(struct wasm_stack *self, size_t capacity)
```

Comment:
Something to consider: self vs. stack is losing information, and editors in C aren't going to associate self with the C namespace it is associated with, `wasm_stack`. This means that a reader might need to "scroll up" to recall the object type/namespace being operated on. The upside is that someone writes a lot of self or rust code will know what this means. But no-one who develops on sledge does either (beside you). I'm not suggesting to change this, but I don't see a benefit, and I see a minor downside.

Reply:
I guess the main benefit is to communicate that "this is a OOP-style function," but if the self or this aren't meaningful in this way, I might as well roll it back. I've got self in a bunch of places in the repo. I think I'm going to take advantage of the different namespaces for structs and have the "self" argument have the same name as the type. That gives us: 

`static inline int wasm_stack_allocate(struct wasm_stack *wasm_stack, size_t capacity)`